bedrock/bedrock-common/src/main/java/com/nukkitx/protocol/bedrock/BedrockSession.java

```diff
@@ -252,12 +254,14 @@ public void onWrappedPacket(final ByteBuf wrappedData) {
         ByteBuf batched = null;
         try {
             if (this.isEncrypted()) {
-                // Decryption
-                batched = ByteBufAllocator.DEFAULT.directBuffer(wrappedData.readableBytes());
-                this.decryptionCipher.cipher(wrappedData.internalNioBuffer(wrappedData.readerIndex(), wrappedData.writerIndex()),
-                        batched.internalNioBuffer(0, wrappedData.writerIndex()));
+                batched = PooledByteBufAllocator.DEFAULT.directBuffer(wrappedData.readableBytes());
+                ByteBuffer output = ByteBuffer.allocateDirect(wrappedData.readableBytes());
```

Comment:
Using ByteBuffer directly is not how this should be fixed because it's not using Netty's allocators which include the memory pooling etc.

Reply:
yeah I agree, its mainly a workaround for the client crash for the moment. I'm just placing my changes here as I find stuff and will cleanup later. More just a dump in case its helpful until resolved.